source/include/m2minterfaceimpl.h

```diff
@@ -435,6 +441,7 @@ friend class M2MInterfaceFactory;
     M2MConnectionHandler        _connection_handler;
     M2MNsdlInterface            _nsdl_interface;
     M2MSecurity                 *_security;
+    char                        _error_description[64];
```

Comment:
I couldn't find initialization of this buffer from constructor, which is needed as one might call error_description() without actual error happening.
The 64 should also be replaced with a constant, as it is used in few other places and having different values there may have bad results.